projects/packages/post-list/src/class-post-list.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * The Post List Admin Area.
+ *
+ * @package automattic/jetpack-post-list
+ */
+
+namespace Automattic\Jetpack\Post_List;
+
+use Automattic\Jetpack\Assets;
+
+/**
+ * The Post_List Admin Area
+ */
+class Post_List {
+	/**
+	 * The configuration method that is called from the jetpack-config package.
+	 */
+	public static function configure() {
+		$post_list = self::get_instance();
+		$post_list->register();
+	}
+
+	/**
+	 * Initialize the Post List UI.
+	 *
+	 * @return Post_List Post_List instance.
+	 */
+	public static function get_instance() {
+		return new Post_List();
+	}
+
+	/**
+	 * Sets up Post List action callbacks if needed.
+	 */
+	public function register() {
+		if ( ! did_action( 'jetpack_on_posts_list_init' ) ) {
+			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+			add_action( 'in_admin_footer', array( $this, 'create_app_root_element' ) );
+
+			/**
+			 * Action called after initializing Post_List Admin resources.
+			 *
+			 * @since $$next-version$$
+			 */
+			do_action( 'jetpack_on_posts_list_init' );
+		}
+	}
+
+	/**
+	 * Enqueue scripts depending on the post-list query var.
+	 *
+	 * @param string $hook Page hook.
+	 */
+	public function enqueue_scripts( $hook ) {
+		if ( 'edit.php' === $hook ) {
+			$plugin_path = Assets::get_file_url_for_environment( './index.js', './index.js', __FILE__ );
+
+			wp_enqueue_script(
+				'jetpack_posts_list_ui_script',
+				$plugin_path,
+				array(),
+				gmdate( 'YW' ),
+				true
+			);
+
+			wp_enqueue_style(
+				'jetpack_posts_list_ui_style',
+				plugin_dir_url( __DIR__ ) . './src/style.css',
+				array(),
+				gmdate( 'YW' )
+			);
+
+			wp_style_add_data(
+				'jetpack_posts_list_ui_style',
+				'rtl',
+				plugin_dir_url( __DIR__ ) . 'build/style.rtl.css'
+			);
+
+			wp_set_script_translations( 'jetpack_posts_list_ui_script', 'jetpack' );
+
+			add_action( 'admin_footer', array( $this, 'print_post_data' ) );
+		}
+	}
+
+	/**
+	 * Outputs a JSON blob to the global `wp_admin_posts` variable, for use
+	 * by the JS application
+	 */
+	public function print_post_data() {
+		global $wp_query;
+
+		if ( ! post_type_supports( $wp_query->query['post_type'], 'thumbnail' ) ) {
+			return;
+		}
+
+		$post_data = array_map(
+			function ( $post ) {
+				$thumbnail = Post_Thumbnail::get_post_thumbnail( $post );
+				return array(
+					'id'             => $post->ID,
+					'type'           => $post->post_type,
+					'featured_image' => $thumbnail,
```

Comment:
Do we want / need to escape the value here?

Reply:
The $thumbnail values should be clean as the only "user input" data we're using is the scraped attachment ID from the post. We cast that ID as an INT and then use that ID to look up all other values from the DB.

But to be cautious, I've added escapes to the $thumbnail values before returning the array in class-post-thumbnail.php.

```
return array(
	'id'    => esc_attr( $image_id ),
	'url'   => esc_url( $image_url ),
	'thumb' => esc_url( $image_thumb ),
	'alt'   => esc_attr( $image_alt ),
);
```